src/transaction.js

```diff
@@ -101,13 +101,17 @@ class Transaction {
         }
         if (assetMetadataHash !== undefined && assetMetadataHash.length !== 0) {
             if (typeof(assetMetadataHash) === 'string') {
-                const encoded = Buffer.from(assetMetadataHash);
-                if (encoded.byteLength !== ASSET_METADATA_HASH_LENGTH) {
-                    throw Error("assetMetadataHash must be a " + ASSET_METADATA_HASH_LENGTH + " byte Uint8Array or string.");
-                }
-                assetMetadataHash = new Uint8Array(encoded);
-            } else if (assetMetadataHash.constructor !== Uint8Array || assetMetadataHash.byteLength !== ASSET_METADATA_HASH_LENGTH)
+                assetMetadataHash = new Uint8Array(Buffer.from(assetMetadataHash));
+            }
+            
+            if (assetMetadataHash.constructor !== Uint8Array || assetMetadataHash.byteLength !== ASSET_METADATA_HASH_LENGTH) {
                 throw Error("assetMetadataHash must be a " + ASSET_METADATA_HASH_LENGTH + " byte Uint8Array or string.");
+            }
+
+            if (assetMetadataHash.every(value => value === 0)) {
+                // if hash contains all 0s, omit it
+                assetMetadataHash = undefined;
+            }
         } else {
             assetMetadataHash = undefined;
         }
```

Comment:
Since the code conditioning on the value of `assetMetadaataHash`, consider having as a sequence of if statements and avoid the `else`s

Reply:
I think this section of code already has the minimum number of `else`s.